chalicelib/checks/wfr_checks.py

```diff
@@ -240,27 +242,49 @@ def line_count_test(connection, **kwargs):
+            # we have a few different dictionaries of steps to check output from in linecount_dicts.py
+            # the proband-only and family workflows have the same steps, so we assign the proband_SNV_dict
+            if 'Proband-only' in metawfr_meta['title'] or 'Family' in metawfr_meta['title']:
+                steps_dict = proband_SNV_dict
+            # trio has novoCaller, so it has a separate dictionary of steps
+            elif 'Trio' in metawfr_meta['title']:
+                steps_dict = trio_SNV_dict
+            # cnv/sv is a completely different pipeline, so has a many different steps
+            elif 'CNV' in metawfr_meta['title']:
+                steps_dict = CNV_dict
+            # if this is run on something other than those expected MWFRs, we want an error.
+            else:
+                e = 'Unexpected MWF Title: '+metawfr_meta['title']
+                action_logs['error'].append(str(e))
                 continue
+            # this calls check_lines from cgap-pipeline pipeline_utils check_lines.py (might get moved to generic repo in the future)
+            # will return TRUE or FALSE if all pipeline steps are fine, or if there are any that do not match linecount with their partners, respectively
+            linecount_result = check_lines(metawfr_uuid, my_auth, steps=steps_dict, fastqs=fastqs_dict)
+            #want an empty dictionary if no overall_qcs, or a dictionary of tests and results if there are items in the overall_qcs list
+            overall_qcs_dict = {qc['name']: qc['value'] for qc in metawfr_meta.get('overall_qcs', [])}
             overall_qcs_dict['linecount_test'] = 'PASS' if linecount_result else 'FAIL'
+            # turn the dictionary back into a list of dictionaries that is properly structured (e.g., overall_qcs: [{"name": "linecount_test", "value": "PASS"}, {...}, {...}])
             updated_overall_qcs = [{'name': k, 'value': v} for k, v in overall_qcs_dict.items()]
-            ff_utils.patch_metadata({'overall_qcs': updated_overall_qcs}, metawfr_uuid, key=my_auth)
-            if linecount_result:
-                action_logs['metawfrs_that_passed_linecount_test'].append(metawfr_uuid)
-            else:
-                action_logs['metawfrs_that_failed_linecount_test'].append(metawfr_uuid)
+            try:
+                ff_utils.patch_metadata({'overall_qcs': updated_overall_qcs}, metawfr_uuid, key=my_auth)
+                if linecount_result:
+                    action_logs['metawfrs_that_passed_linecount_test'].append(metawfr_uuid)
+                else:
+                    action_logs['metawfrs_that_failed_linecount_test'].append(metawfr_uuid)
+            except Exception as e:
+                action_logs['error'].append(str(e))
+                break
         except Exception as e:
-            action_logs['error'] = str(e)
+            action_logs['error'].append(str(e))
             break
     action.output = action_logs
     action.status = 'DONE'
```

Comment:
you could add some logic so that status = 'ERROR' if errors get appended to the log. helpful when a user is searching previous actions.

Reply:
done